site/oauth2-examples-keycloak.md

```diff
@@ -0,0 +1,129 @@
+2. Access management rest api
+3. Access AMQP protocol
+
+## Prerequisites to follow this guide
+
+* Docker
+* make
+
+## Deploy Key Cloak
+
+First, deploy **Key Cloak**. It comes preconfigured with all the required scopes, users and clients.
+```
+make start-keycloak
+```
+**Key Cloak** comes configured with its own signing key. And the [rabbitmq.config](https://github.com/rabbitmq/rabbitmq-oauth2-tutorial/tree/main/conf/keycloak/rabbitmq.config)
+used by `make start-keycloak` is also configured with the same signing key.
+
+To access KeyCloak management interface go to http://0.0.0.0:8080/ and enter `admin` as username and password.
+
+There is a dedicated **KeyCloak realm** called `Test` configured as follows:
+* We configured an [rsa](http://0.0.0.0:8080/admin/master/console/#/realms/test/keys) signing key
+* And a [rsa provider](http://0.0.0.0:8080/admin/master/console/#/realms/test/keys/providers)
+* And three clients: `rabbitmq-client-code` for the rabbitmq managament ui, `mgt_api_client` to access via the
+management api and `producer` to access via AMQP protocol.
+
+
+## Start RabbitMQ
+
+Supporting new types of OAuth 2 servers is currently under development.
+There are two alternatives. You can run directly from source:
+1. git clone rabbitmq/rabbitmq-server
+2. git checkout oidc-integration
+3. `gmake run-broker PLUGINS="rabbitmq_management rabbitmq_auth_backend_oauth2" RABBITMQ_CONFIG_FILE=<root folder of the tutorial>/conf/keycloak/rabbitmq.config`
+
+Or from docker:
+
+<pre class="lang-bash">
+export MODE=keycloak
+make start-rabbitmq
+</pre>
+
+
+## Access Management api
+
+Access the management api using the client [mgt_api_client](http://0.0.0.0:8080/admin/master/console/#/realms/test/clients/c5be3c24-0c88-4672-a77a-79002fcc9a9d) which has the scope [rabbitmq.tag:administrator](http://0.0.0.0:8080/admin/master/console/#/realms/test/client-scopes/f6e6dd62-22bf-4421-910e-e6070908764c)
+
+<pre class="lang-bash">
+make curl-keycloak url=http://localhost:15672/api/overview client_id=mgt_api_client secret=LWOuYqJ8gjKg3D2U8CJZDuID3KiRZVDa
+</pre>
+
+## Access AMQP protocol with PerfTest
+
+To test OAuth2 authentication with AMQP protocol we are going to use RabbitMQ PerfTest tool which uses RabbitMQ Java Client.
+First we obtain the token and pass it as a parameter to the make target `start-perftest-producer-with-token`.
+
+<pre class="lang-bash">
+make start-perftest-producer-with-token PRODUCER=producer TOKEN=$(bin/keycloak/token producer kbOFBXI9tANgKUq8vXHLhT6YhbivgXxn)
+</pre>
+
+**NOTE**: Initializing an application with a token has one drawback: the application cannot use the connection beyond the lifespan of the token. See the next section where we demonstrate how to refresh the token.
+
+## Access AMQP protocol with Pika
+
+This section is about testing Oauth2 authentication with AMQP protocol and with Pika library. And more specifically, we
+are demonstrating how to refresh a token on a live AMQP connections.
+
```

Comment:
Reword this: "This section is about testing Oauth2 authentication with AMQP protocol and with Pika library. And more specifically, we are demonstrating how to refresh a token on a live AMQP connections.

TO THIS:

In the following information, Oauth2 authentication is tested with the AMQP protocol and the Pika library. These tests specifically demonstrate how to refresh a token on a live AMQP connection.